### PR TITLE
feat: Layered.ts can give placement to layers

### PR DESCRIPTION
**Pull request by jbrundage**
Add DonutSummary.ts
Add samples/chart/Pie/DonutSummary.js
Added ability to center value and labels to SummaryC.ts
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

### CODE REVIEW

**Comment by jbrundage** [comment]
image

**Comment by jbrundage** [comment]
@GordonSmith this is a mashup of widgets (like we talked about the other day)

I would say that this is nearly production ready... though I haven't done a thorough testing as I'm not sure what your take is on my approach to the layer placement in `Layered.ts`

File: packages/chart/src/DonutSummary.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { Pie, SummaryC } from "@hpcc-js/chart";
```
Comment (GordonSmith) [inline packages/chart/src/DonutSummary.ts]: Should be:
`import { Pie} from "./Pie";`

File: packages/chart/src/DonutSummary.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import { HTMLWidget } from "@hpcc-js/common";
+import { Layered } from "@hpcc-js/layout";
```
Comment (GordonSmith) [inline packages/chart/src/DonutSummary.ts]: I don't think chart should have a dependency on Layout...

File: packages/chart/src/DonutSummary.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    summaryValue(): number;
+    summaryValue(_: number): this;
+    summaryValue_exists(): boolean;
+}
+DonutSummary.prototype.publish("summaryText", "", "string", "Subtext in centered summary widget");
+DonutSummary.prototype.publish("summaryValue", null, "number", "Value of the centered summary widget (defaults to sum of pie values when empty)");
+DonutSummary.prototype.publishProxy("paletteID", "_donut", "paletteID");
+DonutSummary.prototype.publishProxy("tooltipValueFormat", "_donut", "tooltipValueFormat");
+
+// TODO: Is there a simple way to expose all of the _donut publish properties here?
```
Comment (GordonSmith) [inline packages/chart/src/DonutSummary.ts]: I like the concept - but adding layout / layered and a Canvas widget is very heavy approach, when you could simply add two common/Text widgets.
Also with the Text widgets you can set their sizes and they will clip nicely with "..."s when needed.

**Comment by jbrundage** [comment]
@GordonSmith it looks like comms tests are causing the unit tests to fail

**Pull request merged**